main.cpp

```diff
@@ -3,28 +3,97 @@
+#define DEBUG false
 
 int main() {
-    if(isModeAvailable(WIN))g_color_mode=WIN;
-    else if(isModeAvailable(ANSI))g_color_mode=ANSI;
-    else g_color_mode=NO;
-    srand (time(NULL));
-    int c;
-    cout << "Wpisz rozmiar generowanej planszy lub 0, zeby pobrac plansze z pliku: ";
-    cin >> c;
-    cout << "Pobrano " << c << '\n';
+    size_t wybor;
     try {
-        if(c) {
-            generateMap(c,c);
-            loop();
-        } else {
+        int takasobiezmienna=1;
+        if(DEBUG) {
+            cout << "type \'c\' to break\n";
+            char ch;
+            while(1) {
+              ch=getch();
+              if(ch=='c')break;
+              cout<<(unsigned int)ch<<' ';
+            }
+        }
+        calibrate();//TODO: wsadzić to do menu
+        if(DEBUG) {
+            cout<<"Naciskaj kolejne klawisze strzalek, zeby przetestowac kalibracje.\n";
+            while(1) {
+                cout<<getEvent()<<' ';
+            }
+        }
+        if(isModeAvailable(WIN))g_color_mode=WIN;
+        else if(isModeAvailable(ANSI))g_color_mode=ANSI;
+        else g_color_mode=NO;
+        srand (time(NULL));
+        do{
+    clear_screen();
+    cout<<"Witaj w naszej amatorskiej grze \"Labirynt\"."<<endl<<endl<<endl;
+    cout<<"+---------MENU---------+"                   <<endl;
+    cout<<"| Wybierz:             |"                   <<endl;
+    cout<<"| 1. - Kampania        |"                   <<endl;
+    cout<<"| 2. - Losowa mapa     |"                   <<endl;
+    cout<<"| 3. - Wyswietl Tworcow|"                   <<endl;
+    cout<<"| 4. - Wyjdz           |"                   <<endl;
+    cout<<"+----------------------+"                   <<endl;
+
+    cout<<"Wybieram: ";
+    cin>>wybor;
+if(wybor==1||wybor==2)
+{
+
+    if(wybor==2)
+    {
+        size_t cordy1, cordy2;
+        cout<<"Wybierz rozmiar mapy: "<<endl;
+        cout<<"Podaj szerokosc: ";cin>>cordy1;
+        cout<<"Podaj wysokosc: ";cin>>cordy2;
+        generateMap(cordy1,cordy2);
+        loop();
+    }
+    else if(wybor==1)
+    {
             size_t i = (size_t)-1;
             while(doEnd(++i)) {
                 getMap();
                 loop();
             }
-        }
-    }catch(const char * c) {
+
+    }
+clear_screen();
+cout<<"Gratulacje !!! Wygrales"<<endl;
+getch();
+}
+    else if(wybor==3)
+    {
+        clear_screen();
+        cout << "---------TWORCY---------" <<endl;
+        cout << "Nircek aka Mrcc Znepva" << endl;
+        cout << "MrDarkness19"           << endl;
+        cout << "Crackenhoff"            << endl;
+        cout << "GummyBear"              << endl;
+        cout << "Bialas"                 << endl;
+        getch();
+    }
+    else if(wybor==4)
+    {
+        clear_screen();
+        cout<<"Nastapi wyjscie z programu"<<endl;
+        getchar();
+        return 0;
+    }
+    else
+    {
+        cout<<"Nie znaleziono takiej opcji..."<<endl;
+        getch();
+    }
+
+    }while(true);
```

Comment:
wiem, że się czepiam, ale wiesz, że zamiast `do{}while(true);` możesz zrobić `while(1){}` lub `for(;;)`? krótszy zapis i każdy bd wiedział, że to jest normalna pętla nieskończona, a nie jakiś do...while